src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -0,0 +1,158 @@
+        }
+        case CalendarData_DayNames:
+        {
+            NSArray *standaloneWeekdaySymbols = [dateFormat standaloneWeekdaySymbols];
+            NSString *arrayToString = [[standaloneWeekdaySymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_AbbrevDayNames:
+        {
+            NSArray *shortStandaloneWeekdaySymbols = [dateFormat shortStandaloneWeekdaySymbols];
+            NSString *arrayToString = [[shortStandaloneWeekdaySymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_MonthNames:
+        {
+            NSArray *standaloneMonthSymbols = [dateFormat standaloneMonthSymbols];
+            NSString *arrayToString = [[standaloneMonthSymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_AbbrevMonthNames:
+        {
+            NSArray *shortStandaloneMonthSymbols = [dateFormat shortStandaloneMonthSymbols];
+            NSString *arrayToString = [[shortStandaloneMonthSymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_SuperShortDayNames:
+        {
+            NSArray *veryShortStandaloneWeekdaySymbols = [dateFormat veryShortStandaloneWeekdaySymbols];
+            NSString *arrayToString = [[veryShortStandaloneWeekdaySymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_MonthGenitiveNames:
+        {
+            NSArray *monthSymbols = [dateFormat monthSymbols];
+            NSString *arrayToString = [[monthSymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_AbbrevMonthGenitiveNames:
+        {
+            NSArray *shortMonthSymbols = [dateFormat shortMonthSymbols];
+            NSString *arrayToString = [[shortMonthSymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        case CalendarData_EraNames:
+        case CalendarData_AbbrevEraNames:
+        {
+            NSArray *eraSymbols = [dateFormat eraSymbols];
+            NSString *arrayToString = [[eraSymbols valueForKey:@"description"] componentsJoinedByString:@"||"];
+            return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
+        }
+        default:
+            assert(false);
+            return NULL;
+    }
+}
```

Comment:
Could we bring the 
```
NSString *arrayToString = [[<nsarray_name> valueForKey:@"description"] componentsJoinedByString:@"||"];
return arrayToString ? strdup([arrayToString UTF8String]) : NULL;
```
Outside the switch case? It would make this much shorter and more clear that a bunch of these cases only differ in terms of the `[dateFormat <calendar data type>`

Reply:
Yes that's a good idea, will refactor.